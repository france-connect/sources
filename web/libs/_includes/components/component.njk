{% macro component(name, params = {}) %}
  {% include "components/" + name + ".njk" ignore missing %}
{% endmacro %}