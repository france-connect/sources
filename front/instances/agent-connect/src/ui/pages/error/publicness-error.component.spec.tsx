import { render } from '@testing-library/react';
import { useMediaQuery } from 'react-responsive';

import { PublicnessErrorComponent } from './publicness-error.component';

describe('PublicnessErrorComponent', () => {
  it('should have call useMediaQuery with params', () => {
    // when
    render(<PublicnessErrorComponent />);

    // then
    expect(useMediaQuery).toHaveBeenCalledTimes(1);
    expect(useMediaQuery).toHaveBeenCalledWith({ query: '(min-width: 768px)' });
  });

  it('should match the snapshot for a desktop viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(<PublicnessErrorComponent />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot for a tablet viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(false);
    // when
    const { container } = render(<PublicnessErrorComponent />);

    // then
    expect(container).toMatchSnapshot();
  });
});
