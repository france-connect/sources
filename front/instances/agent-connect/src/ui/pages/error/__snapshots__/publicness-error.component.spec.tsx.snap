// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicnessErrorComponent should match the snapshot for a desktop viewport 1`] = `
<div>
  <div
    class="page fr-m-auto fr-mt-8w"
  >
    <h2
      class="title fr-text--bold fr-px-2w text-center"
    >
      Accès impossible
    </h2>
    <div
      class="fr-mx-auto fr-my-4w fr-px-2w fr-text--lg"
    >
      <p
        class="fr-mb-4w"
      >
        Ce service est réservé aux agents de la Fonction Publique.
      </p>
      <p
        class="fr-mb-4w"
      >
        Vous êtes rattaché à une organisation de droit privé. Vous ne pouvez pas y avoir accès.
      </p>
    </div>
  </div>
</div>
`;

exports[`PublicnessErrorComponent should match the snapshot for a tablet viewport 1`] = `
<div>
  <div
    class="page fr-m-auto fr-mt-4w"
  >
    <h2
      class="title fr-text--bold fr-px-2w text-left"
    >
      Accès impossible
    </h2>
    <div
      class="fr-mx-auto fr-my-4w fr-px-2w fr-text--lg"
    >
      <p
        class="fr-mb-4w"
      >
        Ce service est réservé aux agents de la Fonction Publique.
      </p>
      <p
        class="fr-mb-4w"
      >
        Vous êtes rattaché à une organisation de droit privé. Vous ne pouvez pas y avoir accès.
      </p>
    </div>
  </div>
</div>
`;
